internal/services/cdn/cdn_frontdoor_rule_set_association_resource.go

```diff
@@ -0,0 +1,245 @@
+		return fmt.Errorf("creating %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceCdnFrontDoorRuleSetAssociationRead(d, meta)
+}
+
+func resourceCdnFrontDoorRuleSetAssociationRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	id, err := parse.FrontDoorRuleSetAssociationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if err := validateRuleSetsAssociation(d, meta, id, false); err != nil {
+		return err
+	}
+
+	routeRuleSetAssociations, _, err := getRouteRuleSetProperties(d, meta, id)
+	if err != nil {
+		return err
+	}
+
+	// NOTE: I am pulling the values directly from the route resource because if everything
+	// worked correctly the state of this resource should match that state of the associated
+	// route resource...
+	d.Set("cdn_frontdoor_route_id", parse.NewFrontDoorRouteID(id.SubscriptionId, id.ResourceGroup, id.ProfileName, id.AfdEndpointName, id.AssociationName).ID())
+	d.Set("cdn_frontdoor_rule_set_ids", routeRuleSetAssociations)
+
+	return nil
+}
+
+func resourceCdnFrontDoorRuleSetAssociationUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	id, err := parse.FrontDoorRuleSetAssociationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// lock the route and rule set resource for update...
+	locks.ByID(cdnFrontDoorRouteResourceType)
+	defer locks.UnlockByID(cdnFrontDoorRouteResourceType)
+
+	locks.ByID(cdnFrontDoorRuleSetAssociationResourceType)
+	defer locks.UnlockByID(cdnFrontDoorRuleSetAssociationResourceType)
+
+	if d.HasChange("cdn_frontdoor_rule_set_ids") {
+		// make sure the route and the rule sets exist and are associated with the route...
+		if err := updateRuleSetsAssociations(d, meta, id, "updating", "waiting for the update of"); err != nil {
+			return err
+		}
+	}
+
+	return resourceCdnFrontDoorRuleSetAssociationRead(d, meta)
+}
+
+func resourceCdnFrontDoorRuleSetAssociationDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	// since you are deleting the resource remove all
+	// rule set associations on the route
+	id, err := parse.FrontDoorRuleSetAssociationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// lock the route and rule set resource for delete...
+	locks.ByID(cdnFrontDoorRouteResourceType)
+	defer locks.UnlockByID(cdnFrontDoorRouteResourceType)
+
+	locks.ByID(cdnFrontDoorRuleSetAssociationResourceType)
+	defer locks.UnlockByID(cdnFrontDoorRuleSetAssociationResourceType)
+
+	// Check to see if the route still exists and grab its properties...
+	_, props, err := getRouteRuleSetProperties(d, meta, id)
+	if err != nil {
+		return err
+	}
+
+	// call set directly to skip all route and rule set validation...
+	if err := setRouteRuleSetAssociations(d, meta, id, nil, props, "deleting", "waiting for the deletion of"); err != nil {
+		return err
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func validateRuleSetsAssociation(d *pluginsdk.ResourceData, meta interface{}, id *parse.FrontDoorRuleSetAssociationId, isCreate bool) error {
```

Comment:
as this is doing a get, could we explain the reasoning for not letting the api handle it?

Reply:
Fixed.